kyo-core/shared/src/main/scala/kyo/Meter.scala

```diff
@@ -225,16 +221,123 @@ object Meter:
 
                 def close(using Frame): Boolean < IO =
                     Kyo.foreach(meters)(_.close).map(_.exists(identity))
+
+                def closed(using Frame): Boolean < IO =
+                    Kyo.foreach(meters)(_.closed).map(_.exists(identity))
             end new
         }
 
-    private def offer[A](n: Int, chan: Channel[A], v: A)(using Frame): Unit < IO =
-        Loop.indexed { idx =>
-            if idx == n then Loop.done
+    abstract private class Base(permits: Int)(using initFrame: Frame, allow: AllowUnsafe) extends Meter:
```

Comment:
New `Meter` implementation to avoid the issue with JCTools rounding up the capacity to the next power of two. Instead of managing the permits with a channel, the new approach uses an atomic and a queue for waiters. It's inspired in Twitter Util's `AsyncSemaphore` I [worked on](https://github.com/twitter/util/blob/17c8187d948846756fabcdbb2ff6838b3665968b/util-core/src/main/scala/com/twitter/concurrent/AsyncSemaphore.scala#L1).